src/dependency/DependencyLoader.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include "dependency/DependencyLoaderInterface.h"
+#include "api/ApiInterface.h"
+#include "windows/WinApiInterface.h"
+
+namespace UKControllerPlugin {
+    namespace Dependency {
+
+        /*
+            A class that is responsible for loading dependencies
+            off of the filesystem.
+        */
+        class DependencyLoader : public DependencyLoaderInterface
+        {
+            public:
+                DependencyLoader(
+                    UKControllerPlugin::Windows::WinApiInterface& filesystem
+                );
+
+                nlohmann::json LoadDependency(std::string key , nlohmann::json defaultValue) noexcept override;

```

Comment:
What's the reason behind suppressing the exceptions?

Reply:
Not suppressing as such, this just guarantees that my function will not throw any exceptions - that is, I catch any exceptions from the JSON parser and return defaultValue :)